internal/transport/transport_test.go

```diff
@@ -179,3 +182,57 @@ func TestTimeseries(t *testing.T) {
 		}
 	})
 }
+
+func TestNewFetchConfig(t *testing.T) {
+	t.Parallel()
+
+	testURL, err := url.Parse("https//api.test.com/")
+	if err != nil {
+		t.Fatalf("error parsing url: %v", err)
+	}
+
+	testClient := &web.Client{}
+
+	req := &config.Request{
+		Endpoint:    "/test",
+		Method:      "POST",
+		RateLimiter: rate.NewLimiter(rate.Every(time.Second), 2),
+		Query: map[string]string{
+			"hey": "ho",
+			"foo": "bar",
+		},
+	}
+
+	expected := &web.FetchConfig{
+		Method:      req.Method,
+		URL:         testURL,
+		C:           testClient,
+		RateLimiter: req.RateLimiter,
+	}
+
```

Comment:
Could we put this in a for loop so that we can add new conditions as we run into them?

```go
for _, tcase := range []struct {
    req *config.Request
    expected *web.FetchConfig
} {
    {
        req: &config.Request{
		Endpoint:    "/test",
		Method:      "POST",
		RateLimiter: rate.NewLimiter(rate.Every(time.Second), 2),
		Query: map[string]string{
			"hey": "ho",
			"foo": "bar",
		},
	}, 
        expected: &web.FetchConfig{
		Method:      req.Method,
		URL:         testURL,
		C:           testClient,
		RateLimiter: req.RateLimiter,
	},
    } ,
} {
    // Put your comparison logic here
}
```